sound/soc/intel/common/soc-acpi-intel-adl-match.c

```diff
@@ -209,6 +209,25 @@ static const struct snd_soc_acpi_link_adr adl_sdca_3_in_1[] = {
 	{}
 };
 
+static const struct snd_soc_acpi_link_adr adl_sdw_rt1316_link12_rt714_link0[] = {
+	{
+		.mask = BIT(1),
+		.num_adr = ARRAY_SIZE(rt1316_1_group1_adr),
+		.adr_d = rt1316_1_group1_adr,
+	},
+	{
+		.mask = BIT(2),
+		.num_adr = ARRAY_SIZE(rt1316_2_group1_adr),
+		.adr_d = rt1316_2_group1_adr,
+	},
+	{
+		.mask = BIT(0),
+		.num_adr = ARRAY_SIZE(rt714_0_adr),
+		.adr_d = rt714_0_adr,
+	},
```

Comment:
it may be a better idea to list the entries by link, i.e. with BIT(0) first. That would be different from the name but it'd help detect any inconsistencies faster IMHO.

Reply:
that's a good point @bardliao, I forgot about this detail.
In those cases we should add a comment that the order is by-design and required by the machine driver.